### PR TITLE
Mention that you need to configure max upload size in reverse proxy as well

### PR DESCRIPTION
**Pull request by aaronraimist**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: changelog.d/10122.doc [hunk 1]
```diff
@@ -0,0 +1 @@
+Mention in `homeserver.yaml` that you may need to configure max upload size in your reverse proxy. Contributed by @aaronraimist.
```
Comment (anoadragon453) [inline changelog.d/10122.doc:1]:
```suggestion
Mention in the sample homeserver config that you may need to configure max upload size in your reverse proxy. Contributed by @aaronraimist.
```

File: synapse/config/repository.py [hunk 2]
```diff
@@ -248,6 +248,10 @@ def generate_config_section(self, data_dir_path, **kwargs):
 
         # The largest allowed upload size in bytes
         #
+        # If you are using a reverse proxy you will also need to set this value in
+        # your reverse proxy's config. See
```
Comment (anoadragon453) [inline synapse/config/repository.py:252]:
```suggestion
        # If you are using a reverse proxy you may also need to set this value in
        # your reverse proxy's config. See
```

**Review by anoadragon453 — COMMENTED** [review]
Definitely something a lot of people forget, although I realise that the reverse proxy documentation only specifies instructions on upping the max for nginx. Looking around, I think nginx may be the only webserver which actually limits upload size out of the box...?

I wonder if we should say "notably nginx" here, or something along those lines.

**Comment by aaronraimist** [comment]
According to https://httpd.apache.org/docs/2.0/mod/core.html#limitrequestbody Apache seems to default to unlimited. Caddy 1 seems to default to unlimited https://caddy.its-em.ma/v1/docs/limits though I can't find any documentation confirming that is also true for Caddy 2. Not sure on any others.

**Comment by aaronraimist** [comment]
If Nginx is the only one then https://github.com/matrix-org/synapse/issues/6485 can probably be closed

**Review by anoadragon453 — APPROVED** [review]
Thanks for checking -- the current wording sounds fine to me!

**Pull request merged**